sway/input/cursor.c

```diff
@@ -39,19 +36,42 @@ static void cursor_send_pointer_motion(struct sway_cursor *cursor,
 			};
 
 			if (wlr_box_contains_point(&box, cursor->x, cursor->y)) {
-				surface = xsurface->surface;
-				sx = cursor->x - box.x;
-				sy = cursor->y - box.y;
-				wlr_seat_pointer_notify_enter(seat, surface, sx, sy);
-				wlr_seat_pointer_notify_motion(seat, time, sx, sy);
-				return;
+				*surface = xsurface->surface;
+				*sx = cursor->x - box.x;
+				*sy = cursor->y - box.y;
```

Comment:
If I recall correctly, xwayland surface x/y coordinates are already in layout coordinates. Is that correct?